pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +701,781 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
+        f"/*[local-name()='extension']/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"/descendant::*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/@base",
+        f".//*[local-name()='complexType' and @name='{target_name}']"
+        f"//*[local-name()='simpleContent']"
+        f"/*[local-name()='extension']/*/*/*/@base"
+    ]
+
+    for query in other_queries:
+        result = get_base_type(query)
+        if result:
+            base_type = result[0]
+            return follow_base_type(base_type)
+
+    return None
+
+
+def scrape_namespaces(xsd_url):
+    """
+    Fetch and parse an XSD file from a given URL to extract namespace declarations.
+
+    Parameters:
+    xsd_url (str): The URL of the XSD file to be fetched and parsed.
+
+    Returns:
+    dict: A dictionary containing the namespace declarations found in the XSD file.
+
+    Raises:
+    ValueError: If the XSD file cannot be retrieved (HTTP status code is not 200).
+    """
+    # Fetch XSD content from the URL
+    response = requests.get(xsd_url)
+    if response.status_code != 200:
+        # Handle error if XSD file cannot be retrieved
+        raise ValueError(f"Failed to fetch XSD file from URL: {xsd_url}")
+
+    # Parse the XSD content
+    tree = etree.fromstring(response.content)
+
+    # Extract namespace declarations
+    namespaces = tree.nsmap
+
+    return namespaces
+
+
+def get_creation_date(file_path):
+    """
+    Returns the creation date of a file in ISO 8601 format.
+
+    Args:
+    file_path (str): The path to the file.
+
+    Returns:
+    str: The creation date of the file in ISO 8601 format.
+    """
+    if platform.system() == 'Windows':
+        # On Windows, use os.path.getctime() to get the creation time
+        creation_time = os.path.getctime(file_path)
+    else:
+        # On Unix-based systems, try to get the birth time
+        stat = os.stat(file_path)
+        try:
+            creation_time = stat.st_birthtime
+        except AttributeError:
+            # Fallback to the last modification time if birth time is not available
+            creation_time = stat.st_mtime
+
+    # Convert the creation time to a datetime object
+    dt_object = datetime.fromtimestamp(creation_time)
+
+    # Return the creation date in ISO 8601 format
+    return dt_object.isoformat()
+
+
+def load_yaml_file(yaml_file):
+    """
+    Load and parse a YAML file.
+
+    Parameters:
```

Comment:
Also it seems like you switched from "Args:" to "Parameters:". Mark has made the argument that "Parameters:" works better because it agrees with the way Sphinx formats HTML documentation, so feel free to go ahead and change your old "Args:" to "Parameters:".